src/main/scala/com/github/unchama/seichiassist/subsystems/gacha/application/actions/LotteryOfGachaItems.scala

```diff
@@ -0,0 +1,68 @@
+package com.github.unchama.seichiassist.subsystems.gacha.application.actions
+
+import cats.effect.Sync
+import com.github.unchama.seichiassist.subsystems.gacha.GachaAPI
+import com.github.unchama.seichiassist.subsystems.gacha.domain.bukkit.GachaPrize
+import com.github.unchama.seichiassist.subsystems.gacha.domain.{GachaPrizeId, GachaProbability}
+import com.github.unchama.seichiassist.subsystems.gacha.subsystems.gachaprizefactory.bukkit.StaticGachaPrizeFactory
+
+import scala.annotation.tailrec
+
+/**
+ * ガチャアイテムの抽選を行う作用
+ */
+
+trait LotteryOfGachaItems[F[_]] {
+
+  def lottery: F[Vector[GachaPrize]] = lottery(1)
```

Comment:
適当に `implicit protected def F: Functor[F]` とかをスコープ内に置いて、

```suggestion
  def runLotteryOnce: F[GachaPrize] = lottery(1).map(resultVec => {
    assert(result.length == 1)
    result.head
  })
```

等とすると良さそう